noir-projects/noir-protocol-circuits/crates/types/src/abis/nullifier_key_validation_request.nr

```diff
@@ -7,71 +7,74 @@ use crate::{
 };
 
 struct NullifierKeyValidationRequest {
-    public_key: GrumpkinPoint,
-    secret_key: GrumpkinPrivateKey,
+    master_nullifier_public_key: GrumpkinPoint,
+    app_nullifier_secret_key: Field, // not a GrumpkinScalar because it's output of poseidon2
 }
 
 impl Eq for NullifierKeyValidationRequest {
     fn eq(self, request: NullifierKeyValidationRequest) -> bool {
```

Comment:
Yes sure. But we need to be prioritizing because we are still basically in the prototyping phase so it does not make sense to spend much time on little issues like this. In this case it's not externally facing and it's not that confusing since the functions are simple. It's just ugly 💩